### ISSUE TITLE
potential ALSA fix?

### ISSUE THREAD

**Issue by JonathanReeve** [droserasprout/lenovo-yoga-c930-linux#5]
I've been trying to debug audio on my C930, too. I had a long chat with some ALSA folks, but didn't get around to running the set of hda verb commands as described in the bottom of the thread here: 

https://mailman.alsa-project.org/pipermail/alsa-devel/2018-November/142369.html

Could you give them a try and tell me what works? I would myself, but NixOS is a very different OS.

**Comment by csfalcione** [comment]
Sorry for the late reply. I'm running Linux Mint 19.3 (kernel 5.3.0-40) and alsa-info already reports that Node 0x17 power setting and actual are D0 (i.e. without running the hda verb commands). The subwoofers under the palm rest do not function.

**Comment by JonathanReeve** [comment]
Right, I'm seeing that now, too. Any idea how to debug this? I'm reading in
some places that we can try to just connect random pins using hdajackretask
until something starts to work.

**Comment by csfalcione** [comment]
I wish I had more insight into how to debug further. I'm just a lowly CS undergrad with very little knowledge of alsa and audio.
I spent some time last Fall fiddling with hdajackretask, albeit non-exhaustively, to no avail.
Maybe it's also worth mentioning that I have the c940, no the c930. I figure their audio setup is the same, since reviews only mention differing CPU/GPU.

**Comment by JonathanReeve** [comment]
There's a lot more detail on this issue [here at this kernel bug](https://bugzilla.kernel.org/show_bug.cgi?id=205755). It's a step in the right direction, at least.

**Comment by NumberOneGit** [comment]
I'm curious whether anyone here has tried using Qemu/passthrough to capture the CORB commands.  A number of other ALC298 computers have solved this problem awhile back using this technique and full audio works (other than internal microphone) running Windows 10 in Qemu on Ubuntu after downloading and installing Lenovo's drivers and enabling proper passthrough, so I assume all the proper hda verb commands are passed through as well.  I've been focusing more on trying to get the internal mic to work in Qemu before trying to figure out how to patch the kernel with the new hda verbs, but there are plenty of threads out there for those who are knowledgeable in Linux/Qemu/kernel patching.  I still have a huge learning curve, but am working on it.  Others more experienced may be able to sort this out pretty quickly.  If anyone still has active interest and they haven't already read the same threads I have I can post some links.

**Comment by NumberOneGit** [comment]
https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1851518/+index?comments=all
https://bugzilla.kernel.org/show_bug.cgi?id=189331
https://jcs.org/2018/11/12/vfio
